x-pack/solutions/observability/plugins/observability/public/pages/util/synthetics_alert_params_to_kql.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, type KueryNode, nodeBuilder, toKqlExpression } from '@kbn/es-query';
+import type { SyntheticsMonitorStatusRuleParams } from '@kbn/response-ops-rule-params/synthetics_monitor_status';
+import type { TLSRuleParams } from '@kbn/response-ops-rule-params/synthetics_tls';
+
+const MS_TO_DAYS = 1000 * 60 * 60 * 24;
+
+const supportedAlertParams = [
+  'tags',
+  'locations',
+  'monitorIds',
+  'monitorTypes',
+  'projects',
+] as const;
```

Comment:
Do we have this array somewhere in Synthetics? To make sure we are in sync if something changes there.

Reply:
Ok, I am now referencing the response-ops params and procedurally generating this list of supported params:

630b3d1502f9bcad9d3b77e7c8dba437d147f317

It is less pretty, but more reliable. Drawback is we lost the `as const`-ability of creating specific keys for typing (the params are now typed as `string`), but the plus side is it will reflect any of the rest of the supported params besides the ones we handle specifically in calling code.

Not ideal, but I think it's better than hard-coding the keys like you pointed out.